File: backend-elevator/app/get_docker_ip.py
import subprocess

def get_docker_ip() -> str:
    wsl = subprocess.run(['wsl', 'ip', 'addr'], capture_output=True, text=True)
    lista = [x.strip() for x in wsl.stdout.split('\n')]
    ip = [x for x in lista if 'inet' in x and 'global eth0' in x][0].split(' ')[1].replace('/20','')
    return ip

File: backend-doors/src/utils/getDockerIp.ts
import { execSync } from 'child_process';

export function getDokcerIp(): string {
    try{
        const output = execSync('wsl ip addr', { encoding: 'utf-8' });
        const lines = output.split('\n').map(l => l.trim());
        const inetLine = lines.find(l => l.includes('inet') && l.includes('global eth0'));

        if(!inetLine){
            console.error('Não foi possível encontrar o IP do REDIS.');
            return '';
        }

        const ip = inetLine.split(' ')[1].replace(/\/.*/, '');
        return ip;
    } catch (err) {
        console.error('Erro ao obter o IP do WSL:', err);
        return '';
    }
}